bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/util/TransferStreamingOutput.java

```diff
@@ -24,23 +24,25 @@
 
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.io.output.CountingOutputStream;
+import org.apache.commons.lang3.StringUtils;
 import org.commonjava.indy.IndyMetricsNames;
 import org.commonjava.indy.core.bind.jaxrs.metrics.IndyMetricsBindingsNames;
 import org.commonjava.indy.measure.annotation.IndyMetrics;
 import org.commonjava.indy.measure.annotation.Measure;
 import org.commonjava.indy.measure.annotation.MetricNamed;
+import org.commonjava.maven.galley.model.Transfer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class TransferStreamingOutput
     implements StreamingOutput
 {
 
-    private InputStream stream;
+    private final Transfer transfer;
 
-    public TransferStreamingOutput( final InputStream stream )
+    public TransferStreamingOutput( final Transfer transfer )
     {
-        this.stream = stream;
+        this.transfer = transfer;
```

Comment:
I think the transfer.lockRead() should actually happen in the ctor, so we don't have a potentially large gap in which some other user could delete the file after Indy thinks it sent it back to the user but before the user actually gets it. This was happening in previous versions of TSO, where I think we were passing in the transfer instance like this. We can pass it in, but we need to lock for reading ASAP when we do...and doing it here allows us to unlock it in a finally block in the write method.

Reply:
Actually we don't have a lockRead() in Transfer interface, but only a lockWrite(). I've used the lockWrite() instead, but not sure if that makes sense.
And meanwhile, the Transfer.lockWrite() is not implemented due to the underline PartylineCacheProvider.lockWrite() and FastLock.lockWrite() are not implemented(same to lockRead())